### PR TITLE
MR-3435 updateRate updateContract

### PR DESCRIPTION
**Pull request by haworku**
## Summary
Add `updateRate` and `updateContract` postgres handlers

#### Related issues
https://github.com/Enterprise-CMCS/managed-care-review/pull/1876

#### Test cases covered
Added tests for `insertRate` `updateRate` `findRateWithHistory`

## QA guidance
I would appreciate if someone could look at the circular dependency issue I was seeing in zod - `rateTypes.ts` file. I tried pulling things out into separate files (pulled out definition of "revisions" into its own schema that shared back to rates and contracts)  and didn't see this resolving things. but maybe you will think of a way to do this I'm not thinking of.

### CODE REVIEW

**Comment by haworku** [comment]
@macrael @JasonLin0991 I stubbed out some  future docs on contract and rates 

- types and naming (@macrael you have mentioned this a couple times - can you take the lead on filling that out? I threw in a skeleton
- change history (I can take lead on this - realizing the assuming behind this pattern isn't documented anywhere, now seems like a good time since we have new find with history functions). 

Are there any other docs on this stuff folks are planning on writing? Lets get stubs up so anyone working can add to them as part of PRs.

**Review by macrael — APPROVED** [review]
Looks good to me I 'll take a look at the docs and the import loop tomorrow. I do think we should test the related tables on update to make sure we aren't re-creating things

File: services/app-api/src/postgres/contractAndRates/insertContract.ts [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
     ContractType as PrismaContractType,
 } from '@prisma/client'
 import type { ContractType } from '../../domain-models/contractAndRates'
 import { parseContractWithHistory } from './parseContractWithHistory'
 import { includeFullContract } from './prismaSubmittedContractHelpers'
+import type { ContractFormEditable } from './updateDraftContract'
 
-type InsertContractArgsType = {
+
+type InsertContractArgsType = ContractFormEditable & {
```
[the remark this answers is not in the document]
Comment (haworku) [reply services/app-api/src/postgres/contractAndRates/insertContract.ts:12]: Yes- I thought it also might be beneficial for testing purposes and then thinking more couldn't really see a reason to restrict it.

File: services/app-api/src/postgres/contractAndRates/updateDraftContract.ts [hunk 2]
```diff
@@ -46,12 +115,43 @@ async function updateDraftContract(
...
-                contractType: formData.contractType,
+                populationCovered: populationCovered,
+                programIDs: programIDs,
+                riskBasedContract: riskBasedContract,
+                submissionType: submissionType,
+                submissionDescription: submissionDescription,
+                contractType: contractType,
+                contractExecutionStatus,
+                contractDocuments: {
+                    create: contractDocuments,
```
[the remark this answers is not in the document]
Comment (haworku) [reply services/app-api/src/postgres/contractAndRates/updateDraftContract.ts:126]: I will add this test! @macrael Should I do this for anything else like state contacts?

[the remark this answers is not in the document]
Comment (macrael) [reply services/app-api/src/postgres/contractAndRates/updateDraftContract.ts:126]: probably, contacts are interesting, do they have an ID that comes in from the API? I wonder if we need to like drop/replace them on update?

**Comment by haworku** [comment]
Addressed small issues in [8001c64](https://github.com/Enterprise-CMCS/managed-care-review/pull/1876/commits/8001c6415bdd113324b4e3eddd6cec6f85933991) - will merge this and open new PR with more expansive tests.

**Pull request merged**